fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ExpressionStatisticCalculator.java

```diff
@@ -156,121 +156,294 @@ public ColumnStatistic visitCall(CallOperator call, Void context) {
+                    double squareMinValue;
+                    double squareMaxValue = Math.max(minValue * minValue, maxValue * maxValue);
+                    if (minValue < 0 && maxValue < 0 || minValue >= 0 && maxValue >= 0) {
+                        squareMinValue = Math.min(minValue * minValue, maxValue * maxValue);
+                    } else {
+                        squareMinValue = 0;
+                    }
+                    minValue = squareMinValue;
+                    maxValue = squareMaxValue;
+                    break;
+                case FunctionSet.RADIANS:
+                    // π = 180°, so the ratio is 57.3
+                    minValue = minValue / 57.3;
+                    maxValue = maxValue / 57.3;
+                    break;
+                case FunctionSet.RAND:
+                case FunctionSet.RANDOM:
+                    minValue = 0;
+                    maxValue = 1;
+                    break;
+                case FunctionSet.NEGATIVE:
+                    double negativeMinValue = -maxValue;
+                    double negativeMaxValue = -minValue;
+                    minValue = negativeMinValue;
+                    maxValue = negativeMaxValue;
+                    break;
+                case FunctionSet.POSITIVE:
+                case FunctionSet.FLOOR:
+                case FunctionSet.DFLOOR:
+                case FunctionSet.CEIL:
+                case FunctionSet.CEILING:
+                case FunctionSet.ROUND:
+                case FunctionSet.DROUND:
+                case FunctionSet.TRUNCATE:
+                    // Just use the input's statistics as output's statistics
+                    break;
                 default:
                     return ColumnStatistic.unknown();
             }
+
+            final double averageRowSize;
+            if (callOperator.getType().isIntegerType() || callOperator.getType().isFloatingPointType()
+                    || callOperator.getType().isDateType()) {
+                averageRowSize = callOperator.getType().getTypeSize();
+            } else {
+                averageRowSize = columnStatistic.getAverageRowSize();
+            }
+            return new ColumnStatistic(minValue, maxValue, columnStatistic.getNullsFraction(), averageRowSize,
+                    distinctValue);
         }
 
         private ColumnStatistic binaryExpressionCalculate(CallOperator callOperator, ColumnStatistic left,
                                                           ColumnStatistic right) {
-            double distinctValues = Math.max(left.getDistinctValuesCount(), right.getDistinctValuesCount());
+            final double minValue;
+            final double maxValue;
             double nullsFraction = 1 - ((1 - left.getNullsFraction()) * (1 - right.getNullsFraction()));
+            double distinctValues = Math.max(left.getDistinctValuesCount(), right.getDistinctValuesCount());
+            double averageRowSize = callOperator.getType().getTypeSize();
```

Comment:
Add time functions like date_add? I think time function is more important than math function, many user depend on time function to filter data

Reply:
The following functions are added
1. date_add
2. datediff
3. timediff
4. date_sub